### ISSUE TITLE
Document user.toml usage

### ISSUE THREAD

**Issue by baumanj** [habitat-sh/habitat#4964]
According to https://www.habitat.sh/docs/using-habitat/#using-a-user-toml-file, the
proper location is `/hab/user/myservice/config/user.toml`, but @bixu reports that only `/hab/svc/myservice/user.toml` works. There has been confusion about which location works or is correct, so the documentation should be checked against the code and updated.

**Comment by reset** [comment]
@baumanj it was moved from `/hab/svc/{service}/user.toml` to another location at some point but I believe this location is not in line with the intended layout of the Habitat filesystem. @fnichol and I spoke about this back in November when we discovered the change.

@fnichol this might be a good time to address this since it's becoming a _real thing_ if we officially document it.

**Comment by reset** [comment]
Right, so I guess we'll just need to discuss if we should fix those docs and the code or what

**Comment by krnowak** [comment]
The work was done in this commit: https://github.com/habitat-sh/habitat/commit/a20498a7e8c30e6802136b305f9ad7b205b97aa9

But note that the commit message is slightly wrong because I forgot to update it. We basically check if `user.toml` exists first in a `/hab/user/myservice/config` directory and if it does not, we fall back to the old directory `/hab/svc/myservice`. So it is the other way around than what is being written in the message. Also, only `/hab/user/myservice/config/user.toml` is being watched for changes.

Note that the supervisor determines the path of `user.toml` only at the startup of the service. So if the service is running and is using `/hab/svc/myservice/user.toml`, the service will not pick up and will not start watching the changes in the config from `/hab/user/myservice/config/user.toml` if the file there is created at some point later. There are tests checking these scenarios (`ducp` and `rucp` are "deprecated user config path" and "recommended user config path", respectively): https://github.com/habitat-sh/habitat/blob/85fcde682ca2bab01e9174bb75cc5364319c7369/components/sup/src/manager/service/config.rs#L847-L925

**Comment by bixu** [comment]
@arashzandi ☝️

**Comment by eeyun** [comment]
As  note from triage today - it's likely this behavior will need to change in the near term, if it does then we should update these docs. If we aren't going to deprecate the original path then the deprecation warning should get removed and we should update that original behavior to add the file watch in. Either way there is some discovery work that needs to happen here.
